README.txt
INSTALL
==========

1. Extract module into /sites/all/modules folder.
2. Enable "Path Breadcrumbs" and "Path Breadcrumbs UI" on /admin/modules page.


USAGE
==========

Go to /admin/structure/path-breadcrumbs and create new breadcrumb.

@TODO: provide better documentation and show demo.


CREDITS
=========

Module was developed by Maslouski Yauheni (http://drupalace.ru).
Module development was not sponsored by anyone. It was created for the love of Drupal.

Module was ported to backdropcms by Leonid Kolesnik (kleomash).


CHANGELOG.txt
Path Breadcrumbs 7.x-3.x, xxxx-xx-xx
by Spleshka: Added new token current-page:path-menu-trail:pb-join:* which generates breadcrumbs based on path bits and menu structure.
by Spleshka: Improved performance with static caching of heavy path_breadcrumbs_load_variant() function.
#2473109 by Fant0m, kalabro, petu: Fixed destination parameter is present but doesn't work during editing breadcrumb.


Path Breadcrumbs 7.x-3.2, 2015-02-04
by kalabro: Added new option to enable Path Breadcrumbs on 403/404 pages.


Path Breadcrumbs 7.x-3.1, 2015-01-01
by Spleshka: Removed static caching from path_breadcrumbs_load_all(), because it is already implemented in ctools_export_load_object().
#2354497 by perusio, Spleshka: Checking if translatable property of object exists.
#1893878 by OnkelTem: Use $language_url just like l() does.
by kalabro: Show error message if user inputs path alias.
#2394337 by ekes: Restrict some percent-encoded sequences in path tokens.


Path Breadcrumbs 7.x-3.0, 2014-06-24
by kalabro: Fixed PHP Warning implode(): Invalid arguments passed in _path_breadcrumbs_menu_token_generate().
#2246181 by maximpodorov, kalabro: Cache empty results to prevent heavy variant lookup process. hook_path_breadcrumbs_view_alter() now works for cached items too.
#2261911 by Nicolas Bouteille, kalabro: Clone translations when cloning Path Breadcrumbs item.
#2290447 by R-H, kalabro: Allow to create empty breadcrumbs (disabling).


Path Breadcrumbs 7.x-3.0-rc2, 2014-02-16
#2178169 by kalabro | audealexandre: Added *:pb-join:none token for empty strings.
#2184389 by maximpodorov: Fixed New Panels module code generates warnings in Path Breadcrumbs.
#2068619 by FreekVR, halefx, kalabro: Fixed PHP Warning Missing argument 3 for ctools argument settings form.
#2189149 by korgik | kalabro: Added option to build breadcrumbs for custom URL via path_breadcrumbs_set_breadcrumb() function.


Path Breadcrumbs 7.x-3.0-rc1, 2014-01-26
------------------------------
#2178565 by kalabro | Fina Bluma: Fixed warnings when using Bootstrap theme.
#2175751 by kalabro | VasilyKraev: Fixed strict warning when using RESTWS module.
#2171769 by kalabro | FortEZ: Fixed Profile2 token not replaced from Ctools User context.


Path Breadcrumbs 7.x-3.0-beta7, 2014-01-08
#2122051 by kalabro | zniki.ru: Fixed Machine name validation error when clone PB item.
#2118329 by kalabro | klim_: Improved URL query support (?&=).
#2066223 by sylus, OnkelTem: Add partial jquery_update-dev support.
#2153755 by bkosborne: Fixed Variable "path_breadcrumbs_decode_entities" not deleted on uninstall.
#2149763 by kalabro, manuelBS | Spleshka: Added 'pb-join' token for node menus, current page menu item and Book hierarchy.
#2160011 by kalabro | VasilyKraev: Added "Configuration Management" module support.


Path Breadcrumbs 7.x-3.0-beta6, 2013-10-25
------------------------------
#2067579 by adamzimmermann, kalabro: Fixed External URLs don't work when using Path Breadcrumbs render function.
#1975160 by kalabro: Added Breadcrumbs internal rendering settings per theme.
#2118329 by kalabro, klim_: Added query symbols support (?&=).


Path Breadcrumbs 7.x-3.0-beta5, 2013-09-29
------------------------------
#1976008 by klonos, Spleshka, kalabro: Renamed "Path breadcrumbs" to "Path Breadcrumbs".
by kalabro: Converted ":parent:" token from list to single item in UI autocomplete.
#2089979 by maximpodorov, kalabro: Fixed multiple PHP "Undefined property" Notices.
#2095631 by oskar_calvo, kalabro: Fixed path breadcrumbs cache doesn't work with translation.
#1954868 by kalabro | abautu, rosberg: Reimplemented and fixed %site context.
by kalabro: Added new access plugins: "Taxonomy: term has children" and "Taxonomy: term depth".


Path Breadcrumbs 7.x-3.0-beta4, 2013-08-06
------------------------------
#2021855 by kalabro | ergow: Fixed double url encoding when using Path breadcrumbs render function.
#1893878 by kalabro | Anthony Goode: Fixed Duplicated path prefix in urls when using Path breadcrumbs render function.
#2051905 by kalabro | operations: Fixed Refreshing i18n strings deletes translated strings.
#2033169 by kalabro: Added new option "URL path cleaning method" with CTools and Pathauto support.
#1929110 by samhassell, VasilyKraev, kalabro: Added new option "Maximum title length" for truncating long strings.
#1959890 by rimen, grient, kalabro: Link/title fields maxlength increased to 256.
By kalabro: Added new "pb-join" token for hierarchical taxonomy breadcrumbs.
#2035365 by lecler, kalabro: Fixed Translations lost after reverting path breadcrumbs.
#2018979 by mmikitka, kalabro: Sort breadcrumbs export by natural order for better file diffing.


Path Breadcrumbs 7.x-3.0-beta3, 2013-06-19
------------------------------
#1958842 by kalabro | VasilyKraev, philsward, Spleshka, das-peter: Fixed Path Breadcrumbs doesn't work in Panels content type.
#2021907 by kalabro: Show contextual links for Path Breadcrumbs variant.
#1954002 by rimen, kalabro: Fixed arguments with underscore not work in tokens. Added raw arguments to autocomplete.
#1954868 by kalabro | abautu, rosberg: Fixed %site context doesn't work.
#1982528 by kalabro: Fixed PDOException because of ctools cache object conflicts.
#1946760 by maximpodorov, Spleshka: Provide API for additional Path Breadcrumbs configuration on 4th step.


Path Breadcrumbs 7.x-3.0-beta2, 2013-04-14
------------------------------
#1959052 by zhangx1a0, kalabro, Spleshka: Fixed Cannot change term page breadcrumbs


path_breadcrumb.api.php
<?php

/**
 * @file
 * Hooks provided by Path Breadcrumbs module.
 */

/**
 * Expose Path Breadcrumbs settings.
 *
 * This hook is called by CTools. For this hook to work, you need
 * hook_ctools_plugin_api(). The values of this hook can be overridden
 * and reverted through the UI.
 */
function hook_path_breadcrumb_settings_info() {
  $path_breadcrumbs = array();

  $path_breadcrumb = new stdClass();
  $path_breadcrumb->api_version = 1;
  $path_breadcrumb->path_id = '1001';
  $path_breadcrumb->machine_name = 'example_breadcrumb';
  $path_breadcrumb->name = 'Example breadcrumb';
  $path_breadcrumb->path = 'node/%node';
  $path_breadcrumb->data = 'a:6:{s:6:"titles";a:3:{i:0;s:51:"1%node:content-type (%node:content-type:node-count)";i:1;s:11:"%node:title";i:2;s:19:"%node:comment-count";}s:5:"paths";a:3:{i:0;s:36:"node/%node:content-type:machine-name";i:1;s:14:"node/%node:nid";i:2;s:6:"<none>";}s:4:"home";i:1;s:12:"translatable";i:0;s:9:"arguments";a:1:{s:4:"node";a:3:{s:8:"position";i:1;s:8:"argument";s:14:"entity_id:node";s:8:"settings";a:1:{s:10:"identifier";s:8:"Node: ID";}}}s:6:"access";a:0:{}}';
  $path_breadcrumb->weight = 0;
  $path_breadcrumb->disabled = 0;

  $path_breadcrumbs[] = $path_breadcrumb;

  return $path_breadcrumbs;
}

/**
 * Respond to saving path_breadcrumbs.
 *
 * This hook is invoked after creating new path_breadcrumbs or updating
 * existing one.
 * @param object $path_breadcrumbs
 *    Object with all necessary information from saving form.
 */
function hook_path_breadcrumbs_save($path_breadcrumbs) {
  // @todo Needs function body.
}

/**
 * Respond to path_breadcrumbs deletion.
 *
 * This hook is invoked before path_breadcrumbs variant is removed from
 * the database.
 *
 * @param object $path_breadcrumbs
 */
function hook_path_breadcrumbs_delete($path_breadcrumbs) {
  // @todo Needs function body.
}

/**
 * Act on a path_breadcrumbs object is preparing for view.
 *
 * This hook is invoked before any token replacement.
 *
 * @param object $path_breadcrumbs
 *    Object with path breadcrumb variant loaded from database.
 * @param array $contexts
 *    Ctools contexts from current URL.
 * @return object $path_breadcrumbs
 */
function path_breadcrumbs_view(&$path_breadcrumbs, $contexts) {
  // @todo Needs function body.
}

/**
 * Alter breadcrumbs generated by _path_breadcrumbs_build_breadcrumbs().
 *
 * This hook is invoked after breadcrumbs were built.
 *
 * @param array $breadcrumbs
 *    Alterable array of build breadcrumbs.
 * @param object $path_breadcrumbs
 *    Unalterable object contains both processed and raw titles and paths.
 * @param array $contexts
 *    Ctools contexts from current URL.
 */
function path_breadcrumbs_view_alter(&$breadcrumbs, $path_breadcrumbs, $contexts) {
  // @todo Needs function body.
}
